src/main/java/org/sagebionetworks/bridge/services/OrganizationService.java

```diff
@@ -213,50 +211,45 @@ public void deleteOrganization(String appId, String identifier) {
+    public void addMember(String appId, String identifier, String userId) {
         checkArgument(isNotBlank(appId));
         checkArgument(isNotBlank(identifier));
-        checkNotNull(accountId);
+        checkArgument(isNotBlank(userId));
         
         CAN_EDIT_MEMBERS.checkAndThrow(ORG_ID, identifier);
         
-        Account account = accountDao.getAccount(accountId)
-                .orElseThrow(() -> new EntityNotFoundException(Account.class));
-        
-        RequestContext context = RequestContext.get();
-        if (!context.isInRole(ADMIN) && account.getOrgMembership() != null) {
-            throw new BadRequestException("Account already assigned to an organization.");
-        }
-        
-        account.setOrgMembership(identifier);
-        accountDao.updateAccount(account);
-        sessionUpdateService.updateOrgMembership(account.getId(), identifier);
+        AccountId accountId = AccountId.forId(appId, userId);
+        accountService.editAccount(accountId, (acct) -> {
+            RequestContext context = RequestContext.get();
+            if (!context.isInRole(ADMIN) && acct.getOrgMembership() != null) {
+                throw new BadRequestException("Account already assigned to an organization.");
+            }
+            acct.setOrgMembership(identifier);
+        });
+        sessionUpdateService.updateOrgMembership(userId, identifier);
     }
     
-    public void removeMember(String appId, String identifier, AccountId accountId) {
+    public void removeMember(String appId, String identifier, String userId) {
         checkArgument(isNotBlank(appId));
         checkArgument(isNotBlank(identifier));
-        checkNotNull(accountId);
+        checkArgument(isNotBlank(userId));
         
         CAN_EDIT_MEMBERS.checkAndThrow(ORG_ID, identifier);
-        
-        Account account = accountDao.getAccount(accountId)
-                .orElseThrow(() -> new EntityNotFoundException(Account.class));
-        
-        // Indicate if caller is trying to remove someone from an org they don't belong to
-        if (account.getOrgMembership() == null || !account.getOrgMembership().equals(identifier)) {
-            throw new BadRequestException("Account is not a member of organization " + identifier);
-        }
-        
-        account.setOrgMembership(null);
-        accountDao.updateAccount(account);
-        sessionUpdateService.updateOrgMembership(account.getId(), null);
+
+        AccountId accountId = AccountId.forId(appId, userId);
+        accountService.editAccount(accountId, (acct) -> {
+            if (acct.getOrgMembership() == null || !acct.getOrgMembership().equals(identifier)) {
+                throw new BadRequestException("Account is not a member of organization " + identifier);
+            }
+            acct.setOrgMembership(null);
+        });
+        sessionUpdateService.updateOrgMembership(userId, null);
```

Comment:
So this no longer throws an EntityNotFoundException, and that seems problematic. We may want to throw an exception when you call editAccount.

Reply:
We need userId here. we are also not throwing ENFE from editAccount()